src/promptflow/promptflow/contracts/types.py

```diff
@@ -25,3 +29,32 @@ class FilePath(str):
     """This class is used to hint a parameter is a file path."""
 
     pass
+
+
+@dataclass
+class AssistantDefinition:
+    """This class is used to define an assistant definition."""
+
+    model: str
+    instructions: str
+    tools: list
+
+    @staticmethod
+    def deserialize(data: dict) -> "AssistantDefinition":
+        return AssistantDefinition(
+            model=data.get("model", ""),
+            instructions=data.get("instructions", ""),
+            tools=data.get("tools", [])
+        )
+
+    def serialize(self):
+        return {
+            "model": self.model,
+            "instructions": self.instructions,
+            "tools": self.tools,
+        }
+
+    def setup_tool_invoker(self, working_dir: Optional[Path] = None):
```

Comment:
`init_tool_invoker()` ? In this case we may not provide working_dir parameter (Directly use the flow level working dir for now)

Reply:
Refined.